### PR TITLE
feat(find_bank_by_bic): add find_bank_by_bic + adjusted docs + test

### PR DESCRIPTION
**Pull request by sassman**
This PR introduces the method `find_bank_by_bic` which is pretty self-explanatory.
Tests are added, docs are updated

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -107,4 +118,24 @@ mod tests {
         assert_eq!(bank.rdh_version, Some("300".to_string()));
         assert_eq!(bank.pin_tan_version, Some("300".to_string()));
     }
+
+    // These tests use the real CSV input data.
+    // As such, the tests might break randomly if the input document changes.
+    // This should happen rarely enough that it's still worthwhile to test with real data, though.
+
+    #[test]
+    fn getting_random_bank_works() {
```
Comment (svenstaro) [inline src/lib.rs:127]: Could you rename this to `get_bank_by_bank_code`?

File: src/lib.rs [hunk 2]
```diff
@@ -107,4 +118,24 @@ mod tests {
...
+        let bank_code = "44761312";
+        let bank = get_bank_by_bank_code(bank_code).unwrap();
+
+        assert_bank_matches(&bank);
+    }
+
+    #[test]
+    fn getting_bank_by_bic() {
```
Comment (svenstaro) [inline src/lib.rs:135]: Could you rename this to `get_bank_by_bic`?

File: src/lib.rs [hunk 3]
```diff
@@ -80,21 +80,32 @@ pub fn get_bank_by_bank_code(bank_code: &str) -> Option<Bank> {
     None
 }
 
+/// Retrieves the bank by it's `bic`
```
Comment (svenstaro) [inline src/lib.rs:83]: `its`

**Review by svenstaro — CHANGES_REQUESTED** [review]
Thanks for the feature! A few minor things. :)

**Comment by svenstaro** [comment]
Also please rebase as we now test code in `README.md` as well. :)

**Comment by svenstaro** [comment]
Also feel free to add a changelog entry for this. :)

File: src/lib.rs (hunk 1 above)
Comment (sassman) [reply src/lib.rs]: Then we would get a name collision, but I propose `get_bank_by_bank_code_test` so like with the `_test` suffix. Would that work for you?

File: src/lib.rs (hunk 2 above)
Comment (sassman) [reply src/lib.rs]: same as above.

**Review by sassman — COMMENTED** [review]
Here you go, rebased and Changelog adjusted.

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,36 +9,46 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased] - ReleaseDate
 
+- Add `get_bank_by_bic` to fetch a bank by its BIC
+
 ## [1.0.6] - 2020-11-09
+
```
Comment (svenstaro) [inline CHANGELOG.md:15]: Please remove these spaces, it throws off my changelog tool. :P

Comment (sassman) [reply CHANGELOG.md:15]: Funny, I've used a code formatter to apply common markdown formatting. Might be worth to fix that :)

Comment (sassman) [reply CHANGELOG.md:15]: but sure.. can do it

**Comment by svenstaro** [comment]
Cool, thanks!

**Comment by svenstaro** [comment]
Would you look to open a follow-up and add this functionality to the CLI as well?

**Pull request merged**